std/assembly/index.d.ts

```diff
@@ -1581,9 +1581,9 @@ declare abstract class TypedArray<T> implements ArrayBufferView {
   /** The lastIndexOf() method returns the last index at which a given element can be found in the typed array, or -1 if it is not present. The typed array is searched backwards, starting at fromIndex. */
   lastIndexOf(searchElement: T, fromIndex?: i32): i32;
   /** Returns copied section of an TypedArray from begin inclusive to end exclusive */
-  slice(begin?: i32, end?: i32): TypedArray<T>;
+  slice(begin?: i32, end?: i32): this;
```

Comment:
The "this" type should be returned only when the body of the function contains "return this", that is, it returns a reference to the current instance. This is not the case with "slice" or "subarray". What is returned is a reference to a new instance

Reply:
As I understand it, this only represents the type, and there is no semantic requirement that the value of this must be returned (it is also not guaranteed).

Reply:
I propose following semantic convention which reflect is it "new instance" (TypedArray<T>) or "the same instance" (this)

Reply:
Just see how TS automatically infer return type in playground:
https://www.typescriptlang.org/play?#code/MYGwhgzhAEBiD29oG8CwAoa0CWA7AZgKYBOAghACoAW2EAFAJQoZZbGEAuArsbtBzQgtoAXwzC8RMhACSuCBzC5ghRs0yt23XtFyEA7nETCx6EUA

Reply:
Ok you convinced me